src/lib.rs

```diff
@@ -77,6 +78,9 @@ impl<T> Gc<T> {
         }
     }
 
+    pub(crate) fn base_ptr_offset(&self) -> usize {
+        self.header().0.get() & !1
```

Comment:
I'm a bit confused by the `&!1` -- maybe because I'm not really sure what this function is supposed to be doing.